### PR TITLE
PB-73: Added geo catalogue and import tool search bar

### PR DESCRIPTION
**Pull request by ltshb**
Now we have a search bar for the geocatalog.

Also improved menu items colors

The active layers and geocatalog menu were always white which make it hard
to differentiate the menu title from the menu content.

[Test link](https://sys-map.dev.bgdi.ch/preview/feat-pb-73-search-geo-catalogue/index.html)

### CODE REVIEW

**Comment by ltclm** [comment]
@ltshb great improvement of the user experience.
one small remark after a few tests with the catalogue search.
especially on bigger catalogues like wms.geo.admin.ch the search is a bit laggy on the first characters. maybe it would be better to activate the search not directly but p.e. after the 3 character has been typed.

**Comment by ltshb** [comment]
> @ltshb great improvement of the user experience. one small remark after a few tests with the catalogue search. especially on bigger catalogues like wms.geo.admin.ch the search is a bit laggy on the first characters. maybe it would be better to activate the search not directly but p.e. after the 3 character has been typed.

@ltclm I tried to improved the performance, can you try and let me know what you think ?

**Comment by ltclm** [comment]
thank you @ltshb it is much better now :+1:

**Review by rebert — APPROVED** [review]
@ltshb I love it. As you mentioned, it would be a good idea when the search bar only appears when a wms (external source) has been loaded.

File: src/modules/menu/components/LayerCatalogue.vue [hunk 1]
```diff
@@ -13,22 +14,73 @@ const { layerCatalogue, compact } = defineProps({
...
+        <div v-if="showSearchBar" class="input-group input-group-sm d-flex p-1">
+            <span id="searchCatalogueIcon" class="input-group-text">
+                <FontAwesomeIcon :icon="['fas', 'search']" />
+            </span>
+            <input
+                ref="searchInput"
+                type="text"
+                class="form-control"
+                :class="{ 'rounded-end': !searchText.length }"
+                :placeholder="$t('search_in_catalogue_placeholder')"
```
Comment (pakb) [inline src/modules/menu/components/LayerCatalogue.vue:61]: please replace `$t` by `i18n.t` and importing it as `const i18n = useI18n()` above 😉

File: src/utils/utils.js
Comment (pakb) [inline src/utils/utils.js:1]: no unit tests to cover this new function? 😉

**Review by pakb — APPROVED** [review]
Very cool! Helps greatly search for things in an external layer provider's list!

File: src/utils/utils.js
Comment (ltshb) [reply src/utils/utils.js:1]: I've introduced component testing in next PR https://github.com/geoadmin/web-mapviewer/pull/591 and do more testing in the PB-88 ticket

File: src/modules/menu/components/LayerCatalogue.vue (hunk 1 above)
Comment (ltshb) [reply src/modules/menu/components/LayerCatalogue.vue:61]: My bad copy paste habits 😉

**Pull request merged**